core/src/main/java/hudson/WebAppMain.java

```diff
@@ -262,6 +265,33 @@ public void joinInit() throws InterruptedException {
     }
 
     /**
+     * Creates the plugin manager to use, if customized.
+     * @return The plugin manager to use or {@code null} to use the default one.
+     */
+    private @CheckForNull PluginManager createPluginManager(ServletContext context, File rootDir) {
+        final String pmClassName = context.getInitParameter(PluginManager.class.getName() + ".className");
```

Comment:
Actually, why not just System.getProperty() ? Is there a specific purpose of such construct?